crates/autopilot/src/run_loop.rs

```diff
@@ -555,7 +561,11 @@ pub fn solve_request(
                     class,
                     app_data: order.data.app_data,
                     signature: order.signature.clone(),
-                    fee_policies: fee_policies.get(&order.metadata.uid).unwrap_or_default(),
+                    fee_policies: fee_policies
+                        .get(&order.metadata.uid)
+                        .cloned()
+                        .map(|policies| policies.into_iter().map(FeePolicy::new).collect_vec())
+                        .unwrap_or_default(),
```

Comment:
nit: if you first unwrap or default you don't need the nested map I believe

Reply:
nested map does conversion from domain to `driver` dto so I don't see how we can avoid it.